app/controllers/concerns/foreman/controller/normalize_vmware_storage_controller_attributes.rb

```diff
@@ -0,0 +1,18 @@
+module Foreman::Controller::NormalizeVmwareStorageControllerAttributes
+  extend ActiveSupport::Concern
+
+  private
+
+  def normalize_vmware_storage_controller_attributes(attrs)
+    ctrls_and_vol = JSON.parse(attrs["controllers"]).
```

Comment:
You can delete here and drop line 13.
```suggestion
    ctrls_and_vol = JSON.parse(attrs.delete("controllers")).
```

Reply:
we need to delete `attrs[controllers]` after we extract nvme and scsi controllers from them, Hence, the deleting part is done on line 13

Reply:
I don't see it used. You use `ctrls_and_vol[:controllers]` later on, but you don't depend on `attrs["controllers"]`.